presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java

```diff
@@ -176,5 +158,30 @@ public void close()
             }
             client = null;
         }
+
+        private void processFailure(Exception e)
+        {
+            if (client != null) {
+                client.getHostLocationHandle().reportFailure(e);
+                this.close();
+            }
+        }
+
+        private PrestoThriftService getClient()
```

Comment:
`getClient()` is a critical piece in implementing this interface, so it was previously in front of all methods.